commit e3d38ef819421ea5b69fd52072da61be30043838
Author: Martin Gräßlin <mgraesslin@kde.org>
Date:   Mon Sep 30 13:01:13 2013 +0200

    Don't show borders for a maximized Laptop client
    
    Maximized windows should not have borders.
    
    REVIEW: 113047

diff --git a/kwin/clients/laptop/laptopclient.cpp b/kwin/clients/laptop/laptopclient.cpp
index 958cd2e..6462bd7 100644
--- a/kwin/clients/laptop/laptopclient.cpp
+++ b/kwin/clients/laptop/laptopclient.cpp
@@ -410,13 +410,13 @@ int LaptopClient::layoutMetric(LayoutMetric lm, bool respectWindowState, const K
         case LM_TitleEdgeRight:
         case LM_BorderLeft:
         case LM_BorderRight:
-            return 4;
+            return (maximizeMode() & MaximizeHorizontal) ? 0 : 4;
 
         case LM_BorderBottom:
-            return mustDrawHandle() ? handleSize : 4;
+            return (maximizeMode() & MaximizeVertical) ? 0 : (isResizable() ? handleSize : 4);
 
         case LM_TitleEdgeTop:
-            return 3;
+            return (maximizeMode() & MaximizeVertical) ? 0 : 3;
 
         case LM_TitleEdgeBottom:
             return 1;
